src/script/interpreter.cpp

```diff
@@ -2061,6 +2062,52 @@ uint256 ComputeTaprootMerkleRoot(Span<const unsigned char> control, const uint25
     return k;
 }
 
+bool VerifyAnnex(const std::vector<unsigned char>& annex_vec, ScriptExecutionData& execdata)
+{
+    CDataStream annex(MakeByteSpan(annex_vec), ANNEX_SER_TYPE, ANNEX_SER_VERSION);
+
+    /* Pop annex tag */
+    uint8_t annex_tag;
+    annex.read(AsWritableBytes(Span{&annex_tag, 1}));
+    assert(annex_tag == ANNEX_TAG);
+
+    uint64_t nLastRecordType = 0;
+    while (!annex.empty()) {
+
+	/* We encode the difference between the previous type (initially 0),
+	 * both minimising the encoding and ensuring a canonical ordering
+	 * for record value. */
+	uint64_t nRecordDeltaLen = 0;
+	annex >> VARINT(nRecordDeltaLen);
+	const uint64_t nRecordType = nLastRecordType + (nRecordDeltaLen >> 7);
+	/* The record length is encoded as a delta. */
+	uint64_t nRecordLength = nRecordDeltaLen & 0x7F;
+	if (nRecordLength == 0x7F) {
+		uint64_t nBuffRecordLength = 0;
+		annex >> VARINT(nBuffRecordLength);
+		nRecordLength += nBuffRecordLength;
```

Comment:
nit: better reads if `nRecordLength = 0x7F + nBuffRecordLength`

Reply:
note this is a bitmask above not an assignment